### PR TITLE
[INTERNAL] Refactoring

### PR DESCRIPTION
**Pull request by matz3**
JIRA: CPOUI5FOUNDATION-355

### CODE REVIEW

File: test/lib/themingParameters/dataUri.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+const assert = require("assert");
+// const sinon = require("sinon");
+
+// tested module
+const themingParametersDataUri = require("../../../lib/themingParameters/dataUri");
+
+describe("themingParameters/dataUri", function() {
+	it("should not add theming parameters when library name is missing", function() {
+		const result = {};
```
Comment (codeworrior) [inline test/lib/themingParameters/dataUri.js:10]: this initializer made me aware of the fact that `dataUri` will fail if the `options` indicate to act, but one of `result.{css,rtl,cssVariables}` is null or undefined.

**Review by codeworrior — COMMENTED** [review]
Overall, the refactoring looks good to me.

Not sure if you would like to modernize the code in general. Some methods still use that = this (instead of arrow functions) and could use async (e.g. cacheTheme).

File: test/lib/themingParameters/dataUri.js (hunk 1 above)
Comment (matz3) [reply test/lib/themingParameters/dataUri.js:10]: Correct. It doesn't fail, but produces a string starting with `undefined` (or `null`).
The contract was that if options are enabled, the results properties must be set. Right now I would stick with that. Handling this case and throwing an error would be a good improvement.

**Comment by matz3** [comment]
Using new language features is a good thing but I didn't focus on changing all the code. Can be done as another follow-up, along with fixing the `max-len` issues.
I only focused on using async/await as it really improves the code structure.

**Comment by matz3** [comment]
Rebased on top of master to also test on Node 16 (see #183)

**Review by codeworrior — APPROVED** [review]
LGTM. Fixing code styles issues later is totally acceptable, same for modernising.

**Comment by matz3** [comment]
Thanks 👍🏻

**Comment by matz3** [comment]
**Note:** I'm using rebase and merge to keep the history and have separate commits for changing the code and moving it to other files.

**Pull request merged**